pkgs/applications/editors/eclipse/plugins.nix

```diff
@@ -437,34 +500,16 @@ rec {
     };
   };
 
-  jsonedit = buildEclipsePlugin rec {
-    name = "jsonedit-${version}";
+  jsonedit = mirrorEclipsePlugins rec {
+    pname = "jsonedit";
     version = "1.1.1";
 
-    srcFeature = fetchurl {
-      url = "https://boothen.github.io/Json-Eclipse-Plugin/features/jsonedit-feature_${version}.jar";
-      sha256 = "0zkg8d8x3l5jpfxi0mz9dn62wmy4fjgpwdikj280fvsklmcw5b86";
-    };
-
-    srcPlugins =
-      let
-        fetch = { n, h }:
-          fetchurl {
-            url = "https://boothen.github.io/Json-Eclipse-Plugin/plugins/jsonedit-${n}_${version}.jar";
-            sha256 = h;
-          };
-      in
-        map fetch [
-          { n = "core"; h = "0svs0aswnhl26cqw6bmw30cisx4cr50kc5njg272sy5c1dqjm1zq"; }
-          { n = "editor"; h = "1q62dinrbb18aywbvii4mlr7rxa20rdsxxd6grix9y8h9776q4l5"; }
-          { n = "folding"; h = "1qh4ijfb1gl9xza5ydi87v1kyima3a9sh7lncwdy1way3pdhln1y"; }
-          { n = "model"; h = "1pr6k2pdfdwx8jqs7gx7wzn3gxsql3sk6lnjha8m15lv4al6d4kj"; }
-          { n = "outline"; h = "1jgr2g16j3id8v367jbgd6kx6g2w636fbzmd8jvkvkh7y1jgjqxm"; }
-          { n = "preferences"; h = "027fhaqa5xbil6dmhvkbpha3pgw6dpmc2im3nlliyds57mdmdb1h"; }
-          { n = "text"; h = "0clywylyidrxlqs0n816nhgjmk1c3xl7sn904ki4q050amfy0wb2"; }
-        ];
-
-    propagatedBuildInputs = [ antlr-runtime_4_7 ];
+    repository = "https://boothen.github.io/Json-Eclipse-Plugin";
+    installableUnits = [
+      { id = "jsonedit-feature.feature.group"; version = version; }
+      { id = "org.antlr.v4.feature.group"; version = "4.7.1"; }
+    ];
+    hash = "sha256:0af5d26lw1yxck5w73wkv57fpjx49pcx58gpflbrbxl3m5pc6js1";
```

Comment:
As mentioned in the root comment of the PR, this change was needed due to peculiarities with the antlr dependency. I think it makes a good case that the new system is easier to use though!